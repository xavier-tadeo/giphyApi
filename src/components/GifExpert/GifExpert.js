import React, { useState } from "react";
import { AddCategory } from "../AddCategory/AddCategory";
import { GifGrid } from "../GifGrid/GifGrid";
import "./GifExpert.scss";

export const GifExpert = ({ defaultCategory = [] }) => {
  const [categories, setCategories] = useState(defaultCategory);

  // const handleAdd = () => {
  //    setCategories([...categories, "Barret de palla"]);
  // setCategories((categories) => [...categories, "Barret"]);
  // };
  return (
    <div className="list">
      <AddCategory setCategories={setCategories} />
      <ol className="list__container">
        {categories.map((categorie) => {
          return <GifGrid categorie={categorie} key={categorie} />;
        })}
      </ol>
    </div>
  );
};
