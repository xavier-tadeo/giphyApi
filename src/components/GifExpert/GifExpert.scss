@import url("https://fonts.googleapis.com/css2?family=Permanent+Marker&display=swap");

.list {
  &__container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    list-style: cambodian;
    color: #0606b1;
    font-family: "Permanent Marker", cursive;
    font-size: 20px;
  }
}
