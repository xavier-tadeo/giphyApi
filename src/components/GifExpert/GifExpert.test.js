import { GifExpert } from "./GifExpert";
import { shallow } from "enzyme";

describe("Given GifExpert component", () => {
  describe("When it render", () => {
    test("It should rendered a divTag and olTag", () => {
      const wrapper = shallow(<GifExpert />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  test("It should the number of category", () => {
    const category = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpert defaultCategory={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(category.length);
  });
});
