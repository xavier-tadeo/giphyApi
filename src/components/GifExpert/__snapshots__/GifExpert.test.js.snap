// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given GifExpert component It should the number of category 1`] = `
<div
  className="list"
>
  <AddCategory
    setCategories={[Function]}
  />
  <ol
    className="list__container"
  >
    <GifGrid
      categorie="One Punch"
      key="One Punch"
    />
    <GifGrid
      categorie="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Given GifExpert component When it render It should rendered a divTag and olTag 1`] = `
<div
  className="list"
>
  <AddCategory
    setCategories={[Function]}
  />
  <ol
    className="list__container"
  />
</div>
`;
